src/chunk_iterator.rs

```diff
@@ -1,20 +1,22 @@
-use std::io;
-use std::io::{BufRead, Read};
+use std::{
+    io,
+    io::{BufRead, Read},
+};
 
 use crate::parser::EncodedChunk;
 
 /// An iterator for reading `chunk_size` bytes off the given `reader`
 pub struct BufferedIterator<T>
 where
-    T: Read,
+    T: BufRead,
 {
     reader: T,
     chunk_size: u64,
 }
 
 impl<T> BufferedIterator<T>
 where
-    T: Read,
+    T: BufRead,
```

Comment:
Can you comment on the impact here? I know Read is "anything that can read", and bufread means "anything that can read with a buffer, so as to be efficient", but it feels like I would want to be "generic" about it?

Am happy taking the change, as it is indeed more akin to what I _mean to write_, but curious of detailed "why".

Reply:
since you don't offer any way to construct or use `BufferedIterator` with T: Read (only with T: BufRead) you might as well restrict it here.

if you had some methods on it that required only `Read`, and those methods were useful in their own right, then it would make sense to relax the trait bound